### PR TITLE
Fix broken logic state for formstep submission

### PR DESCRIPTION
**Pull request by sergei-maertens**
Fixes open-formulieren/open-forms#1494

Given that:
 
 1. a form was in a submittable state
 2. user input in the form step is done in quick succession but the net
    end result is no changes (e.g. radio yes/no toggle to the other
    option and quickly back to the originally selected option)
 
 Then the change handler caused the form submission to be blocked as
 a safeguard against bypassing logic checks.
 
 The logic check in turn shortcutted because the data being compared
 was not different than the originally checked data, so no changes in
 state are suspected.
 
 However, this shortcut/guard clause failed to restore the original
 canSubmit state of the form step, which was set to false by the
 change handler.
 
 This patch ensures that on early-exits of logic checks, the state is
 correctly restored. Additionally, the canSubmit state for aborted
 logic checks now no longer only depends on form validity, but also
 on the previous state. Even though a client-side form is valid,
 it may still be blocked from submission by logic rules.

### CODE REVIEW

**Comment by Bartvaderkin** [comment]
I tested this and while the Next button-state does update to the expected enabled/disabled state there is a small ~1 second window where a race-condition allows that after I select the option that would disable the button I can quickly click the still enabled button and proceed to the next step.

**Comment by sergei-maertens** [comment]
Did some more tweaking and passing around data from older events, I think it's now alright.

It did highlight for me again the complexity of this state management and the need to refactor this to make the code understandable again.

**Review by Bartvaderkin — APPROVED** [review]
I tested the new changes with my reproduction and my earlier issue with the debounce seems fixed.

File: src/components/FormStep.js [hunk 1]
```diff
@@ -395,7 +440,15 @@ const FormStep = ({
...
 
-    dispatch({type: 'BLOCK_SUBMISSION'});
+    // backend logic leads to changes in FormIO configuration, which triggers onFormIOInitialized.
+    // This in turn triggers the onFormIOChange event because the submission data is set
+    // programmatically. Without checking for human interaction, this would block the
+    // submission again for LOGIC_CHECK_DEBOUNCE ms, for the logic check to eventually
+    // be interrupted inside the evaluateFormLogic handler because the data hasn't
+    // changed. We can skip this particular block-unblock cycle by only blocking the
+    // submission because of human input.
+    modifiedByHuman && dispatch({type: 'BLOCK_SUBMISSION'});
```
Comment (Bartvaderkin) [inline src/components/FormStep.js:450]: not a huge fan of this boolean construct over a simple visible if-statements

**Review by SilviaAmAm — APPROVED** [review]

File: src/components/FormStep.js (hunk 1 above)
Comment (sergei-maertens) [reply src/components/FormStep.js]: changed!

**Comment by sergei-maertens** [comment]
Backported via 1bdf03a9d20887113e0cdb9d57b1c255b3d8436f

**Pull request merged**